server/src/main/java/io/crate/planner/operators/CorrelatedJoin.java

```diff
@@ -149,10 +149,10 @@ public LogicalPlan replaceSources(List<LogicalPlan> sources) {
     @Override
     public LogicalPlan pruneOutputsExcept(SequencedCollection<Symbol> outputsToKeep) {
         var toCollect = new LinkedHashSet<>(outputsToKeep);
+        toCollect.remove(selectSymbol);
         selectSymbol.relation().visitSymbols(tree ->
             tree.visit(OuterColumn.class, outerColumn -> toCollect.add(outerColumn.symbol()))
         );
-        toCollect.remove(selectSymbol);
```

Comment:
Removing the `selectSymbol` afterwards causes flakiness in the order of the outputs while column pruning.